cmake/Metal.cmake

```diff
@@ -2,6 +2,29 @@ if(NOT APPLE)
     return()
 endif()
 
+set(BFLOAT_METAL_CODE "
```

Comment:
Why do we have metal code in the .cmake file? Can me move that inside some utils.metal file?

Reply:
This one is not a real kernels, it just checks if host can compile something, this is a pretty standard practice to embed trivial test code in `.cmake` files, see https://github.com/pytorch/pytorch/blob/86b45bde196b976247f2771196f8c1c689dd66b7/cmake/Modules/FindAVX.cmake#L5-L14 for example